src/getThreadList.js

```diff
@@ -1,73 +1,211 @@
+    return {
+      threadID: utils.formatID(t.thread_key.thread_fbid || t.thread_key.other_user_id),
+      name: getThreadName(t),
+      unreadCount: t.unread_count,
+      messageCount: t.messages_count,
+      imageSrc: t.image?t.image.uri:null,
+      emoji: t.customization_info?t.customization_info.emoji:null,
+      color: formatColor(t.customization_info?t.customization_info.outgoing_bubble_color:null),
+      nicknames: mapNicknames(t.customization_info),
+      muteUntil: t.mute_until,
+      participants: formatParticipants(t.all_participants),
+      participantIDs: formatParticipants(t.all_participants).map(participant => participant.userID),
+      adminIDs: t.thread_admins.map(a => a.id),
+      folder: t.folder,
+      isGroup: t.thread_type === "GROUP",
+      // rtc_call_data: t.rtc_call_data, // TODO: format and document this
+      // isPinProtected: t.is_pin_protected, // feature from future? always false (2018-04-04)
+      customizationEnabled: t.customization_enabled, // false for ONE_TO_ONE with Page or ReducedMessagingActor
+      participantAddMode: t.participant_add_mode_as_string, // "ADD" if "GROUP" and null if "ONE_TO_ONE"
+      montageThread: t.montage_thread?Buffer.from(t.montage_thread.id,"base64").toString():null, // base64 encoded string "message_thread:0000000000000000"
+      // it is not userID nor any other ID known to me...
+      // can somebody inspect it? where is it used?
+      // probably Messenger Day uses it
+      reactionsMuteMode: t.reactions_mute_mode,
+      mentionsMuteMode: t.mentions_mute_mode,
+      isArchived: t.has_viewer_archived,
+      isSubscribed: t.is_viewer_subscribed,
+      timestamp: t.updated_time_precise, // in miliseconds
+      // isCanonicalUser: t.is_canonical_neo_user, // is it always false?
+      // TODO: how about putting snippet in another object? current implementation does not handle every possibile message type etc.
+      snippet: t.last_message.nodes[0].snippet,
+      snippetAttachments: t.last_message.nodes[0].extensible_attachment, // TODO: not sure if it works
+      snippetSender: utils.formatID((t.last_message.nodes[0].message_sender.messaging_actor.id || "").toString()),
+      lastMessageTimestamp: t.last_message.nodes[0].timestamp_precise, // timestamp in miliseconds
+      lastReadTimestamp: (t.last_read_receipt.nodes[0]?t.last_read_receipt.nodes[0].timestamp_precise:null), // timestamp in miliseconds
+      cannotReplyReason: t.cannot_reply_reason, // TODO: inspect possible values
+
+      // @Legacy
+      threadType: t.thread_type === "GROUP" ? 2 : 1 // "GROUP" or "ONE_TO_ONE"
+    };
+  });
+}
+
+module.exports = function(defaultFuncs, api, ctx) {
+  return function getThreadList(limit, timestamp, tags, callback) {
+    if (!callback && (utils.getType(tags) === "Function" || utils.getType(tags) === "AsyncFunction")) {
+      callback = tags;
+      tags = [""];
+    }
+    if (utils.getType(limit) !== "Number" || limit%1 !== 0 || limit <= 0) {
```

Comment:
Can you do `isNaN` instead of this modulo trick?

Reply:
Unless this isn't to check if limit is NaN?

Reply:
This is to check if the number is an integer and not a floating point number

Reply:
How about `Number.isInteger()`? (See [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger))

Reply:
How about `Number.isInteger()`? (See [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger))

Reply:
good idea